controlplane/kubeadm/internal/kubeadm_config_map.go

```diff
@@ -43,11 +45,33 @@ const (
 	schedulerKey             = "scheduler"
 )
 
-// kubeadmConfig wraps up interactions necessary for modifying the kubeadm config during an upgrade.
+// KubeadmConfig wraps up interactions necessary for modifying the kubeadm config during an upgrade.
+type KubeadmConfig interface {
+	GetConfigMap() *corev1.ConfigMap
+	RemoveAPIEndpoint(endpoint string) error
+	UpdateKubernetesVersion(version semver.Version) error
+	UpdateImageRepository(imageRepository string) error
+	UpdateEtcdMeta(imageRepository, imageTag string) (bool, error)
+	UpdateCoreDNSImageInfo(repository, tag string) error
+	UpdateAPIServer(apiServer kubeadmv1.APIServer) (bool, error)
+	UpdateControllerManager(controllerManager kubeadmv1.ControlPlaneComponent) (bool, error)
+	UpdateScheduler(scheduler kubeadmv1.ControlPlaneComponent) (bool, error)
```

Comment:
What about having all this method being prefixed with Reconcile, and returning error only in case we cannot reconcile the value to the desired state (and not returning a changed flag)

Reply:
Refer to my comment above, around the use of `changed` flag for one of the methods `UpdateEtcdMeta`.